lib/ansible/module_utils/redfish_utils.py

```diff
@@ -826,6 +826,45 @@ def get_fan_inventory(self):
         result["entries"] = fan_results
         return result
 
+    def get_chassis_power(self):
+        result = {}
+        key = "Power"
+
+        # Get these entries, but does not fail if not found
+        properties = ['Name', 'PowerAllocatedWatts',
+                      'PowerAvailableWatts', 'PowerCapacityWatts',
+                      'PowerConsumedWatts', 'PowerMetrics',
+                      'PowerRequestedWatts', 'RelatedItem', 'Status']
+
+        chassis_power_results = []
+
+        # Go through list
+        for chassis_uri in self.chassis_uri_list:
+            chassis_power_result = {}
+
+            response = self.get_request(self.root_uri + chassis_uri)
+            if response['ret'] is False:
+                return response
+            result['ret'] = True
+            data = response['data']
+            if key in data:
+                response = self.get_request(self.root_uri + chassis_uri +
+                                            "/" + key)
+                data = response['data']
+                if 'PowerControl' in data:
+                    data = data['PowerControl'][0]
+                    for property in properties:
+                        if property in data:
+                            chassis_power_result[property] = data[property]
+                else:
+                    return {'ret': False, 'msg': 'Key PowerControl not found.'}
+                chassis_power_results.append(chassis_power_result)
+            else:
+                return {'ret': False, 'msg': 'Key Power not found.'}
+
+            result['chassis_power'] = chassis_power_results
+            return result
```

Comment:
This is only returning the chassis power from the first chassis. Also, check the format of the result payload being returned and be sure it matches the structure in psu_inventory and fan_inventory (including the "entries" array ).

Reply:
This should be resolved now.

Reply:
Looks like this is not resolved yet. The return statement is within the `for` loop, so it breaks out after one loop. And the results and not within an "entries" array (see get_psu_inventory() for comparison).